src/main/java/net/sf/jabref/gui/groups/TransferableEntrySelection.java

```diff
@@ -39,7 +39,8 @@
     public TransferableEntrySelection(List<BibEntry> list) {
         this.selectedEntries = list;
         selectedEntriesCiteKeys = String.join(",",
-                this.selectedEntries.stream().map(BibEntry::getCiteKey).collect(Collectors.toList()));
+                this.selectedEntries.stream().map(BibEntry::getCiteKeyOptional).filter(key -> key.isPresent())
```

Comment:
I was not fully sure either. But I think that's one of the current limits of the optional stream and ist listed for jdk9...
http://stackoverflow.com/questions/22725537/using-java-8s-optional-with-streamflatmap